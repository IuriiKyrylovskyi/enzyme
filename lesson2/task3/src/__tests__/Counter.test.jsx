import React from "react";
import { shallow } from "enzyme";
import Counter from "../Counter";

describe("Counter", () => {
  let wrappedComponent;

  beforeEach(() => {
    wrappedComponent = shallow(<Counter />);
  });

  it("should display counter with initial value 0", () => {
    expect(wrappedComponent.find(".counter")).toMatchSnapshot();
  });

  it("should display counter value 0", () => {
    expect(wrappedComponent.find(".counter__value").text()).toEqual("0");
  });

  it("should increase counter value on click by 1", () => {
    wrappedComponent.setState({
      counter: 2,
    });
    wrappedComponent.find(".counter__button").at(1).prop("onClick")();

    expect(wrappedComponent.find(".counter__value").text()).toEqual("3");
  });

  it("should decrease counter value on click by 1", () => {
    wrappedComponent.setState({
      counter: 2,
    });

    wrappedComponent.find(".counter__button").at(0).prop("onClick")();

    expect(wrappedComponent.find(".counter__value").text()).toEqual("1");
  });

  it("should decrease counter value on click by 1", () => {
    wrappedComponent.setState({
      counter: 2,
    });
    
    wrappedComponent.find(".counter__button").at(0).prop("onClick")();

    expect(wrappedComponent.find(".counter__value").text()).toEqual("1");
  });

  it("should reset counter value to 0", () => {
    wrappedComponent.setState({
      counter: 2,
    });

    wrappedComponent.find(".counter__value").prop("onClick")();

    expect(wrappedComponent.find(".counter__value").text()).toEqual("0");
  });
});
