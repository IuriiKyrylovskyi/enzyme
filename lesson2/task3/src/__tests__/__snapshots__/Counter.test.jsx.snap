// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter should display counter with initial value 0 1`] = `
<div
  className="counter"
>
  <button
    className="counter__button"
    onClick={[Function]}
  >
    -
  </button>
  <span
    className="counter__value"
    onClick={[Function]}
  >
    0
  </span>
  <button
    className="counter__button"
    onClick={[Function]}
  >
    +
  </button>
</div>
`;
